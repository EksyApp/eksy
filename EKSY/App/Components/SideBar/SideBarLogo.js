import React, {Component} from 'react'
import {Image} from 'react-native'

class SideBarLogo extends Component {

  render() {
    return(
      <Image source={require('../../Images/logoPlaceHolder.png')}/>
    )
  }
}

export default SideBarLogo
