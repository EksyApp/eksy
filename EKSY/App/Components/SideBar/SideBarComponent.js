import React, {Component} from 'react'
import {View, Text} from 'react-native'
import SideBarLogo from '../../Components/SideBar/SideBarLogo'
import {Button} from '../../Components/Common/index'
import * as Theme from '../../Theme/index'
import PropTypes from 'prop-types'
import {UserShape} from "../../Utils/PropTypeShapes";

//Renders the sidebar view that holds buttons to navigate in app
//Go to map closes the sidebar and returns to MainViewComponent
//Settings takes to SettingsComponent
//Add marker takes to AddMarkerComponent
//Login takes to LoginComponent or SignUpComponent
//Admin takes to AdminToolsComponent
export default class SideBarComponent extends Component {

	renderLoginButton() {
		if (!this.props.user) {
			return (
					<Button onPress={this.props.goToLoginScreen}>
						Login
					</Button>
			)
		}
	}

	renderAddMarkerButton() {
		if (this.props.user) {
			return (
					<Button onPress={this.props.goToAddMarker}>
						Add a marker
					</Button>
			)
		}
	}

	renderAdminButton() {
		if (this.props.user && this.props.user.admin) {
			return (
					<Button onPress={this.props.goToAdminTools}>
						Admin
					</Button>
			)
		}
	}

	render() {
		return (
				<View style={styles.menubarStyle}>
					<View style={styles.logoStyle}>
						<SideBarLogo />
					</View>
					<View style={styles.buttonList}>
						<Button onPress={this.props.goToMap}>
							Go to map
						</Button>
						<Button onPress={this.props.goToUserSettings}>
							Settings
						</Button>
						{this.renderAddMarkerButton()}
						{this.renderLoginButton()}
						{this.renderAdminButton()}
					</View>
					{__DEV__ && <Text style={{flex: 1, marginLeft: 20, fontSize: 12}}>debug version</Text>}
				</View>
		)
	}
}

const styles = {
	menubarStyle: {
		flex: 1,
		flexDirection: 'column',
		// Placeholder background color
		backgroundColor: Theme.backgroundColor,
	},
	buttonList: {
		flexDirection: 'column',
		justifyContent: 'space-around',
	},
	logoStyle: {
		// Three blue lines are from the logo picture
	}
}

SideBarComponent.propTypes = {
	user: UserShape,
	goToLoginScreen: PropTypes.func,
	goToAddMarker: PropTypes.func,
	goToAdminTools: PropTypes.func,
	goToMap: PropTypes.func,
	goToUserSettings: PropTypes.func,
}
