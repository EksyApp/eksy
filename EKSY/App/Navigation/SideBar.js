import React, {Component} from 'react'
import {Image, ScrollView, Text, View} from 'react-native'
import {Actions} from 'react-native-router-flux'
import SideBarLogo from './SideBarLogo'
import { Button } from '../Components/Common'
import * as ReduxActions from '../Actions'
import {connect} from 'react-redux'
import * as Theme from '../Theme'

export class SideBar extends Component {

  render() {
    return (
      <View style={styles.menubarStyle}>
        <View style={styles.logoStyle}>
            <SideBarLogo />
        </View>
        <View style={styles.buttonList}>
          <Button onPress={this.props.goToMap}>
            Go to map
          </Button>
          <Button onPress={this.props.goToUserSettings}>
            Settings
          </Button>
          <Button onPress={this.props.goToAddMarker}>
            Add a marker
          </Button>
          <Button onPress={this.props.goToLoginScreen}>
            Login
          </Button>
        </View>
      </View>
    )
  }
}

const styles = {
  menubarStyle: {
    flex: 1,
    flexDirection: 'column',
    // Placeholder background color
    backgroundColor: Theme.backgroundColor,
  },
  buttonList: {
    flexDirection: 'column',
    justifyContent: 'space-around',
  },
  logoStyle: {
    // Three blue lines are from the logo picture
  }
}

const mapStateToProps = (state) => {
  return {

  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    goToMap: () => {
      Actions.mapContainer()
      dispatch(ReduxActions.drawerClose())
    },
    goToUserSettings: () => {
      Actions.userSettings()
      dispatch(ReduxActions.drawerClose())
    },
    goToAddMarker: () => {
      Actions.addMarker()
      dispatch(ReduxActions.drawerClose())
    },
    goToLoginScreen: () => {
      Actions.login()
      dispatch(ReduxActions.drawerClose())
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SideBar)
