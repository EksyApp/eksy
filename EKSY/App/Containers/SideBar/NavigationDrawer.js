import React, {Component} from 'react'
import SideBar from './SideBarContainer'
import {DefaultRenderer} from 'react-native-router-flux'
import * as Actions from '../../Actions/index'
import {SideMenu} from 'react-native-elements'
import {connect} from 'react-redux'
import PropTypes from 'prop-types'

class NavigationDrawer extends Component {
	
	render() {
		const children = this.props.navigationState.children
		return (
				<SideMenu
						isOpen={this.props.drawerOpen}
						menu={<SideBar />}
						onChange={(isOpen) => !isOpen ? this.props.drawerClose() : false}
						disableGestures={this.props.disableGestures}
				>
					<DefaultRenderer navigationState={children[0]} onNavigate={this.props.onNavigate}/>
				</SideMenu>
		)
	}
	
}

NavigationDrawer.propTypes = {
	navigationState: PropTypes.object,
	drawerClose: PropTypes.func,
	drawerOpen: PropTypes.bool,
	disableGestures: PropTypes.bool,
	onNavigate: PropTypes.bool
}

const mapStateToProps = (state) => {
	return {
		drawerOpen: state.ui.drawer.drawerOpen,
		disableGestures: state.ui.drawer.disableGestures
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		drawerClose: () => {
			dispatch(Actions.drawerClose())
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(NavigationDrawer)
