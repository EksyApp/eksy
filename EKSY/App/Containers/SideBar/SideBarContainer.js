import React, {Component} from 'react'
import {Actions} from 'react-native-router-flux'
import * as ReduxActions from '../../Actions/index'
import {connect} from 'react-redux'
import SideBarComponent from "../../Components/SideBar/SideBarComponent";

export class SideBarContainer extends Component {

  render() {
    return (
      <SideBarComponent
          goToMap={this.props.goToMap}
          goUserSettings={this.props.goToUserSettings}
          goToAddMarker={this.props.goToAddMarker}
          goToLoginScreen={this.props.goToLoginScreen}
      />
    )
  }
}

const mapStateToProps = (state) => {
  return {

  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    goToMap: () => {
      Actions.mainViewContainer()
      dispatch(ReduxActions.drawerClose())
    },
    goToUserSettings: () => {
      Actions.userSettings()
      dispatch(ReduxActions.drawerClose())
    },
    goToAddMarker: () => {
      Actions.addMarker()
      dispatch(ReduxActions.drawerClose())
    },
    goToLoginScreen: () => {
      Actions.login()
      dispatch(ReduxActions.drawerClose())
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SideBarContainer)
