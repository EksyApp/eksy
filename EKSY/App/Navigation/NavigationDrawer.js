import React, { Component } from 'react'
import PostOffice from '../lib/PostOffice'
import {Drawer} from 'native-base'
import SideBar from './SideBar'
import {DefaultRenderer} from 'react-native-router-flux'

class NavigationDrawer extends Component {

  constructor(props) {
    super(props)
    this.isOpen = false;
    this.po = new PostOffice();
    this.po.register("drawer", (state) => {this.handleDrawer(state)});
    this.po.getPacket("drawer").open = false;
  }

  openDrawer() {
    this.drawer._root.open()
    this.isOpen = true;
  }

  closeDrawer() {
    this.drawer._root.close();
    this.isOpen = false;
  }

  handleDrawer(state) {
    if (state.open && !this.isOpen) {
      this.openDrawer();
    } else if (!state.open && this.isOpen) {
      this.closeDrawer();
    }
  }

  render() {

      const state = this.props.navigationState;
      const children = state.children;
      return (
        <Drawer
          ref={(ref) => { this.drawer = ref; }}
          content={<SideBar /*navigator={this.navigator}*/ />}
          onClose={() => {
            this.po.getPacket("drawer").open = false;
            this.po.sendPacket("drawer");
          }}
          >
          <DefaultRenderer navigationState={children[0]} onNavigate={this.props.onNavigate} />
        </Drawer>
      );

  }

}

export default NavigationDrawer
