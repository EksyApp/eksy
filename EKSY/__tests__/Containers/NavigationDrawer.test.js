import React from 'react'
import { NavigationDrawer } from '../../App/Containers/SideBar/NavigationDrawer'
import renderer from 'react-test-renderer'

jest.mock('../../App/Containers/SideBar/SideBarContainer', () => {
	return require('react-native').View
})

test('NavigationDrawer renders correctly', () => {
	const tree = renderer.create(
			<NavigationDrawer
					navigationState={{children: {}}}
					drawerClose={() => {}}
					drawerOpen
					disableGestures
					onNavigate={() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});