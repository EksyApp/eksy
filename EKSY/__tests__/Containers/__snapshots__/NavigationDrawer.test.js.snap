// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationDrawer renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "right": 250,
        },
      ]
    }
  >
    <View />
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateX": 500,
          },
        ],
        "width": 750,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID={undefined}
    />
  </View>
</View>
`;
